src/Microsoft.AspNetCore.OData/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using Microsoft.AspNet.OData.Adapters;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Microsoft.AspNet.OData.Formatter.Serialization
+{
+    /// <summary>
+    /// Context information used by the <see cref="ODataSerializer"/> when serializing objects in OData message format.
+    /// </summary>
+    public partial class ODataSerializerContext
+    {
+        private HttpRequest _request;
+        private IUrlHelper _urlHelper;
+
+        /// <summary>
+        /// Gets or sets the HTTP Request whose response is being serialized.
+        /// </summary>
+        public HttpRequest Request
+        {
+            get { return _request; }
+
+            set
+            {
+                _request = value;
+                //TODO: Set InternalRequest.
```

Comment:
Is this TODO cleared by the line preceeding it?

Reply:
Nope, like line 42, another link is needed to set InternalRequest to a new instance of a class that does not exist in the code yet. It will be in the next PR.